ivadomed/training.py

```diff
@@ -233,24 +242,33 @@ def train(model_params, dataset_train, dataset_val, training_params, path_output
                         preds = model(input_samples, metadata)
                     else:
                         preds = model(input_samples)
+                        if model_params["name"] == "HourglassNet":
+                            preds = preds[-1] 
+                            jvis = [batch["gt_metadata"][k][0]["jvis"] for k in range(len(batch["gt_metadata"]))]
+                            jvis = torch.Tensor(jvis).view(len(jvis), -1)
+                            jvis = imed_utils.cuda(jvis, cuda_available, non_blocking=True)
+                            gt_samples = (gt_samples.float(), jvis)
 
                     # LOSS
                     loss = loss_fct(preds, gt_samples)
+                    if model_params["name"] == "HourglassNet":
+                        gt_samples = gt_samples[0]
                     val_loss_total += loss.item()
                     val_dice_loss_total += loss_dice_fct(preds, gt_samples).item()
 
                     # Add frame to GIF
                     for i_ in range(len(input_samples)):
                         im, pr, met = input_samples[i_].cpu().numpy()[0], preds[i_].cpu().numpy()[0], \
                                       batch["input_metadata"][i_][0]
+                        if model_params["name"] == "HourglassNet":                           
+                           pr = np.sum(preds[i_].cpu().numpy(), axis=0) 
```

Comment:
Thanks for your suggestion. 
Well in the pose model we have a prediction for each intervertebral disc, thus, it is multi-channel in to visualize it we need to sum all the predictions. It can be added as a comment in the code.

Reply:
Sorry, it might be too obvious... I understand now. :sweat_smile: